src/ui/Crosshair.js

```diff
@@ -8,8 +8,44 @@ export default class Crosshair extends LineSegments {
     const geometry = createGeometry(0.05);
     const material = createMaterial();
 
-    super(geometry, material);
+    const obj = super(geometry, material);
+
+    this.size = 0.05;
+    this.selected = false;
+    this.timer = 0;
+    this.timeLimit = 30;
+    return obj;
+  }
+
+  update(selector) {
+    // TODO: Rewrite this animation using tweens.
+
+    if (!this.selected && selector.selectEvent && selector.selected) {
+      this.selected = true;
+    }
+    else if (this.selected && this.timer < this.timeLimit) {
+      this.timer += 3;
+      this.size += 0.002;
+      this.updateGeometry(this.size);
+    }
+    else if (this.selected) {
+      this.size = 0.05;
+      this.updateGeometry(this.size);
+      this.selected = false;
+      this.timer = 0;
+    }
   }
+
+  updateGeometry(size) {
```

Comment:
Changing geometry should be avoided. Wouldn't scaling the crosshair have the save effect?